### PR TITLE
Add audit testing for IMA measurement

### PR DESCRIPTION
**Pull request by wnereiz**
Add audit testing for IMA measurement

- Related ticket: https://progress.opensuse.org/issues/48926
- Verification run: 
  - SLE: http://10.67.17.9/tests/691
  - Tumbleweed: http://10.67.17.9/tests/692

### CODE REVIEW

File: lib/main_common.pm [hunk 1]
```diff
@@ -2212,9 +2212,10 @@ sub load_security_tests_mok_enroll {
 
 sub load_security_tests_ima_measurement {
     loadtest "security/ima/ima_setup";
-    loadtest "security/ima/ima_measurement";
-    loadtest "security/ima/ima_kernel_cmdline_template";
-    loadtest "security/ima/ima_kernel_cmdline_hash";
+    #loadtest "security/ima/ima_measurement";
+    #loadtest "security/ima/ima_kernel_cmdline_template";
+    #loadtest "security/ima/ima_kernel_cmdline_hash";
+    loadtest "security/ima/ima_measurement_audit";
```
Comment (lilyeyes) [inline lib/main_common.pm]: the three "#"(s) should be deleted :)

Comment (wnereiz) [reply lib/main_common.pm]: My mistake. It is for debugging. Modified.

File: tests/security/ima/ima_measurement_audit.pm [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        $self->select_serial_terminal;
+
+        # Clear audit log
+        assert_script_run("echo -n '' > /var/log/audit/audit.log");
+
+        ($f->{cmd}) ? assert_script_run($f->{cmd}) : die "Get command failure";
+
+        # We do not check the exact file hash here, but to ensure the audit
+        # record existed
+        assert_script_run("ausearch -m INTEGRITY_RULE |grep '$f->{file}.*hash='");
```
Comment (lilyeyes) [inline tests/security/ima/ima_measurement_audit.pm]: The code LGTM.

But I want to mention the verification of audit log is quite simple,
e.g., there is no separated/double checking for "BPRM_CHECK", "FILE_CHECK", and "MMAP_CHECK".
But I know maybe there is no way for these kind of double check.

One question, there is no tests for type "INTEGRITY_PCR", why?

File: lib/main_common.pm (hunk 1 above)
Comment (lilyeyes) [reply lib/main_common.pm]: Thanks for the update:)

File: tests/security/ima/ima_measurement_audit.pm (hunk 2 above)
Comment (wnereiz) [reply tests/security/ima/ima_measurement_audit.pm]: INTEGRITY_PCR is related to TPM support, which seems not available now for openQA workers (with simulation solution). I consider to add it in future once it become available.

**Pull request merged**